### DISCUSSION TITLE
False Positives In Secret Detection

### DISCUSSION THREAD

**Discussion by behara** [aquasecurity/trivy#4297] [Q&A]
## False Positive In Secret Scanning

When we scan an image with trivy we see false positives in secret scanning detection

## What did you expect to happen?
When i use docker file

FROM alpine:3.16.3
COPY vendor.0dccb716b100d1dd.js.map /

This image shouldn't have false positives.

BTW attached zip file of vendor.0dccb716b100d1dd.js.map [vendorFile.zip](https://github.com/aquasecurity/trivy/files/11232228/vendorFile.zip). Or else you can scan image automation.azurecr.io/secret:fp from docker hub.

## What happened instead?

3 secrets got detected but all are false positives

https://github.com/aquasecurity/trivy/blob/4b36e97dcebb8c2f912cbfc55b04f10ba88b4c72/pkg/fanal/secret/builtin-rules.go#L771
https://github.com/aquasecurity/trivy/blob/4b36e97dcebb8c2f912cbfc55b04f10ba88b4c72/pkg/fanal/secret/builtin-rules.go#L762
https://github.com/aquasecurity/trivy/blob/4b36e97dcebb8c2f912cbfc55b04f10ba88b4c72/pkg/fanal/secret/builtin-rules.go#L420

Also the match content is huge

## Output of run with `-debug`:

```
Please check it content is huge
```

## Output of `trivy -v`:

```
Version: dev
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-04-14 06:09:41.908036711 +0000 UTC
  NextUpdate: 2023-04-14 12:09:41.908036411 +0000 UTC
  DownloadedAt: 2023-04-14 09:56:28.600213492 +0000 UTC

```

**Comment by afdesk** [comment]
@behara 
~~it seems `0dccb716b100d1dd` looks like ID )~~
**UPD** no, my first thought was wrong. this large file contains some characters match the build-in rules.

you can disable these rules for your cases.
https://aquasecurity.github.io/trivy/v0.39/docs/secret/configuration/#disable-rules

**Comment by afdesk** [comment]
i managed to find the first problem place:
![image](https://user-images.githubusercontent.com/19297627/232048598-98ce798d-6349-4f17-92cf-922ed018b47a.png)

**Comment by afdesk** [comment]
the second place:
![image](https://user-images.githubusercontent.com/19297627/232049120-182887d8-5ef7-4475-95b5-cc549e783512.png)
the third place:
![image](https://user-images.githubusercontent.com/19297627/232049195-277a7b73-bc49-47be-abc4-3ba51e68ae69.png)
